packages/input-option/src/InputOption/InputOption.types.ts

```diff
@@ -3,13 +3,48 @@ import { PropsWithChildren } from 'react';
 import { AriaLabelProps } from '@leafygreen-ui/a11y';
 import { DarkModeProps } from '@leafygreen-ui/lib';
 
+const ActionType = {
+  Default: 'default',
+  Destructive: 'destructive',
+} as const;
+
+type ActionType = (typeof ActionType)[keyof typeof ActionType];
+
+export { ActionType };
+
+// InputOption components receive different styling and have different variants available to them
+// Therefore, we need to care about where the component is rendered to determine how to render the component appropriately.
+const RenderedContext = {
+  Form: 'form',
+  Menu: 'menu',
+} as const;
+
+type RenderedContext = (typeof RenderedContext)[keyof typeof RenderedContext];
+
+export { RenderedContext };
+
+const State = {
+  Default: 'default',
+  Hover: 'hover',
+  Highlight: 'highlight',
+  Disabled: 'disabled',
+  Checked: 'checked',
+  Destructive: 'destructive',
+} as const;
+
+type State = (typeof State)[keyof typeof State];
+
+export { State };
+
+export type FormState = Exclude<State, 'destructive'>;
+
 /**
  * TERMINOLOGY
  *
- * `focused`: The element is "focused" via keyboard navigation
- * (Does not mean `:focus`, since input options are not focused in this sense)
+ * `highlighted`: The element is aria-selected or "focused" via keyboard navigation
```

Comment:
This is a little confusing because this says the `element is aria-selected` while `checked` says `The element is selected`.